server-phoenix/test/helios_web/controllers/events_api/events_api_controller_test.exs

```diff
@@ -3,148 +3,151 @@ defmodule HeliosWeb.WebHooks.EventsAPIControllerTest do
+        "thîs is ñot vãlid base64!"
+      )
+      |> post(
+        Routes.events_api_path(conn, :handle, "1", user: "rvrx"),
+        File.read!("test/helios_web/controllers/events_api/example-commit.json")
       )
 
     # Bad Request (Malformed)
     assert conn_1.status == 400
+    assert conn_6.status == 400
     # Not Found (No endpoint at URL)
     assert conn_2.status == 404
     # Created
     assert conn_3.status == 201
     # Already Reported (Duplicate event)
     assert conn_4.status == 208
+    # Unauthorized (Signature could not be verified)
+    assert conn_5.status == 401
   end
 
   test "POST new commit", %{conn: conn} do
+    Repo.insert!(%User{
+      user_name: "rvrx",
+      public_key: File.read!("test/helios_web/controllers/events_api/pub.pem")
+    })
+
     conn =
       conn
       |> put_req_header("content-type", "application/json")
+      |> put_req_header(
+        "http-auth",
+        "cPlQ8bGl/l2CD2/F0vAyqHPv+hnWhyJSWIBaMCALfJAqWNONyk6F2FzVTLrQFwVh8UfMi4wrqwJnSjJ34I+ceFIn1K83BRjUvKMT8z5XU3jFop0+6/DOwq6XW4YV00hf8ypnNFlcIqlXFhe5CQcI0bKvX1h7xhH5NBNro7lVTMEX7br29fTb92OvvM3HULc73SDc9+0dDaNtHsRC+KMqtXRtqCm+kSIjWsXBk6bMUPKzUu+kZeMxNofKhh0ewGTKgASuD+r2mBI/U+7M4/903XkSH6yWrcWNkUF+lbYr2skNO1wVvD/SG82bTj2W9/0zpG8CbSd+68eZHVcQjkwnkw=="
+      )
       |> post(
-        Routes.events_api_path(conn, :handle, "1"),
-        %{
-          event_type: "github_commit",
-          author: "RVRX",
-          event_id: "a26asde67d8f5eaf1f6ba5c57fc3c7d91ac0fd1c"
-        }
+        Routes.events_api_path(conn, :handle, "1", user: "rvrx"),
+        File.read!("test/helios_web/controllers/events_api/example-commit.json")
       )
 
     assert conn.status == 201
     assert length(get_commits()) == 1
   end
 
   test "POST new PR", %{conn: conn} do
+    Repo.insert!(%User{
+      user_name: "rvrx",
+      public_key: File.read!("test/helios_web/controllers/events_api/pub.pem")
+    })
+
     conn =
       conn
       |> put_req_header("content-type", "application/json")
+      |> put_req_header(
+        "http-auth",
+        "wSwi8IC54s6EudNlVP6dV9JznKpZ4F7Ka/p9wWauo5xr2sfLGJVi3+s/J1OZQwbBx3ZpyqP20YNB/cpMTE5u40jDpTlAHe7rc8vLkWjHmA7/8LBECXeGl1msip12D5rw9Auk88hobqRcB+wDJAccGLwCwyUv5ZJ+JZRO7W10/jeJ6mM8kzpntA2GJnwM6SbV/DyZ+vgQOwmAUf6PpnkXnWrK8t2+sKV5azKO1umAkrmEuNCV5ifKuuI1tEq9Lzpt9d5unsh0L/oQySXHBEvWDr/UOwo1VS36GgL+ccYRSwH2FAL2kqsMHJvGj63w9JbfpPL4eTxZXqw5OmyEHjffZQ=="
+      )
       |> post(
-        Routes.events_api_path(conn, :handle, "1"),
-        %{
-          event_type: "github_pr",
-          author: "RVRX",
-          event_id: "726abde67d8f5eaf1f6ba5c57fc3c7d91ac0fd1c"
-        }
+        Routes.events_api_path(conn, :handle, "1", user: "rvrx"),
+        File.read!("test/helios_web/controllers/events_api/example-pr.json")
       )
 
     assert conn.status == 201
     assert length(get_commits()) == 0
     assert length(get_prs()) == 1
   end
 
-  test "POST multiple commits (w/ duplicate IDs) and PRs", %{conn: conn} do
```

Comment:
:thinking: I feel like these were useful tests. If you add the throw away private key, can you sign them programmatically to continue to use them?

Reply:
I kinda felt like they were redundant, but was maybe biased by the chore of not wanting to sign and encrypt anymore payloads. I'll add them back